### PR TITLE
http: converting const to constexpr

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: http: converting const to constexpr
Additional Description:
Minor change to the http utility function, that is more aligned with the Envoy guidelines.

Risk Level: low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Review by jmarantz — DISMISSED** [review]

File: source/common/http/utility.cc [hunk 1]
```diff
@@ -856,7 +856,7 @@ Utility::getLastAddressFromXFF(const Http::RequestHeaderMap& request_headers,
   }
 
   absl::string_view xff_string(xff_header->value().getStringView());
-  static const std::string separator(",");
+  static constexpr absl::string_view separator(",");
   // Ignore the last num_to_skip addresses at the end of XFF.
   for (uint32_t i = 0; i < num_to_skip; i++) {
     const std::string::size_type last_comma = xff_string.rfind(separator);
```
Comment (jmarantz) [inline source/common/http/utility.cc:862]: nit: this should be absl::string_view::size_type.

Comment (adisuissa) [reply source/common/http/utility.cc:862]: Thanks for catching this. Done.

**Review by jmarantz — DISMISSED** [review]

File: source/common/http/utility.cc (hunk 1 above)
Comment (wbpcode) [reply source/common/http/utility.cc:862]: nit: I think there still is a `std::string::size_type` being used.

By the way, does string view contains a `absl::string_view::npos`? If it does then should we use it here?

**Comment by wbpcode** [comment]
A nit comment. Feel free to ping me if you think it doesn't matter. (I have no any strong point to that.)

**Comment by jmarantz** [comment]
Good catch @wbpcode might as well get this right.

File: source/common/http/utility.cc (hunk 1 above)
Comment (adisuissa) [reply source/common/http/utility.cc]: Thanks for catching this!
I probably should to get a new pair of eyeglasses :)
Fixed all places that referenced std::string and did not need to create an explicit string object.

**Review by jmarantz — APPROVED** [review]

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**